sympy/utilities/tests/test_lambdify.py

```diff
@@ -1576,3 +1576,12 @@ def assertAllClose(self, result, *, abstol=1e-15, reltol=1e-15):
 def test_deprecated_set():
     with warns_deprecated_sympy():
         lambdify({x, y}, x + y)
+
+def test_23536_lambdify_cse_dummy():
+
+    f = Function('x')(y)
+    g = Function('w')(y)
+    expr = z + (f**4 + g**5)*(f**3 + (g*f)**3)
+    expr = expr.expand()
+    eval_expr = lambdify(((f, g), z), expr, cse=True)
+    eval_expr((1.0, 2.0), 3.0)
```

Comment:
Why doesn't this test actually check the value returned by the function. It clearly shows the bug:
```python
In [2]: eval_expr((1.0, 2.0), 3.0)
Out[2]: [300.0]
```

Reply:
The bug this PR proposed to fix was the issue reported in #23536, which was about a NameError when evaluating a function proposed from lambdify with the cse flag. There was no reason to test what the output is, only that it no longer raised the error.

The reason this new bug Peter reported is now present is because unit tests were not added when the cse flag was introduced checking the possible input types to lambdify. This line didn't even evaluate before.

Reply:
These are the tests introduced with the cse flag addition (https://github.com/sympy/sympy/blob/master/sympy/utilities/tests/test_lambdify.py#L1580):

```
    cases = [
        Case(
            args=(x, y, z),
            exprs=[
             x + y + z,
             x + y - z,
             2*x + 2*y - z,
             (x+y)**2 + (y+z)**2,
            ],
            num_args=(2., 3., 4.)
        ),
        Case(
            args=(x, y, z),
            exprs=[
            x + sympy.Heaviside(x),
            y + sympy.Heaviside(x),
            z + sympy.Heaviside(x, 1),
            z/sympy.Heaviside(x, 1)
            ],
            num_args=(0., 3., 4.)
        ),
        Case(
            args=(x, y, z),
            exprs=[
            x + sinc(y),
            y + sinc(y),
            z - sinc(y)
            ],
            num_args=(0.1, 0.2, 0.3)
        ),
        Case(
            args=(x, y, z),
            exprs=[
                Matrix([[x, x*y], [sin(z) + 4, x**z]]),
                x*y+sin(z)-x**z,
                Matrix([x*x, sin(z), x**z])
            ],
            num_args=(1.,2.,3.),
            requires_numpy=True
        ),
        Case(
            args=(x, y),
            exprs=[(x + y - 1)**2, x, x + y,
            (x + y)/(2*x + 1) + (x + y - 1)**2, (2*x + 1)**(x + y)],
            num_args=(1,2)
        )
    ]
```

The `exprs` are all lists of expressions and do no include any symbols that have to be dummified.